### PR TITLE
report2idea: validate timestamps in IDEA message before checking rules

### PR DESCRIPTION
**Pull request by cejkato2**
Timestamps in EventTime, DetectTime, Ceasetime should be in interval
(CreateTime - dpast, CreateTime + dfuture).
Currently, dpast is set to 1 and dfuture is set to 0.

Additional tests (in pydoc) were written and doctests.py executes them
during `python setup.py test`.

### CODE REVIEW

File: pycommon/report2idea.py [hunk 1]
```diff
@@ -78,6 +107,59 @@ def signal_h(signal, f):
...
+        return True
+    else:
+	create = parseRFCtime(create)
+    deltapast = create - timedelta(dpast)
+    deltafuture = create + timedelta(dfuture)
+    for t in [et, ct, dt]:
+        if t:
+            # convert timestamp to datetime
+	    t = parseRFCtime(t)
+	    # check allowed interval
```
Comment (vaclavbartos) [inline pycommon/report2idea.py]: Lines 150, 156 and 157 contain tabs instead of spaces.

File: pycommon/report2idea.py [hunk 2]
```diff
@@ -215,6 +297,11 @@ def Run(module_name, module_desc, req_type, req_format, conv_func, arg_parser =
             # Pass the input record to conversion function to create IDEA message
             idea = conv_func(rec, args)
 
+            # Sanity check of timestamps
+            if not check_valid_timestamps(idea):
+                print("Invalid timestamps in skipped message: {0}".format(idea))
+                continue
```
Comment (vaclavbartos) [inline pycommon/report2idea.py]: This will crash if `idea is None`. Move this block below the test for `None` (and maybe add a test with a message which can't be converted so `conv_func` returns `None`).

File: pycommon/report2idea.py (hunk 1 above)
Comment (cejkato2) [reply pycommon/report2idea.py]: fixed in 6c31616

File: pycommon/report2idea.py (hunk 2 above)
Comment (cejkato2) [reply pycommon/report2idea.py]: fixed in c3d5063, thanks for reporting this bug
